diff --git a/lib/index.js b/lib/index.js
index 7d60b01fd385297c5cde1c14df711fd6790ea376..9cb1bee3bd9b9454e4abf5ba1ac701d7dc037de0 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -2504,19 +2504,14 @@ const addNodeBookmark = (ctx, node, pageNumber, registry) => {
         return;
     node.children.forEach((child) => addNodeBookmark(ctx, child, pageNumber, registry));
 };
-const addBookmarks = (ctx, root) => {
-    const registry = {};
-    const pages = root.children || [];
-    pages.forEach((page, i) => {
-        addNodeBookmark(ctx, page, i, registry);
-    });
-};
-
 const render = (ctx, doc) => {
     const pages = doc.children || [];
     const options = { imageCache: new Map(), fieldSets: [] };
-    pages.forEach((page) => renderNode(ctx, page, options));
-    addBookmarks(ctx, doc);
+    const registry = {};
+    pages.forEach((page, index) => {
+        renderNode(ctx, page, options);
+        addNodeBookmark(ctx, page, index, registry);
+    });
     ctx.end();
     return ctx;
 };
